include/memory_allocator.h

```diff
@@ -0,0 +1,61 @@
+#ifndef __MEMORY_ALLOCATOR_H__
+#define __MEMORY_ALLOCATOR_H__
+
+#define MAGIC 0xC0DECAFE
+#define ALIGNMENT 8
+
+TAILQ_HEAD(mb_list, mem_block);
+TAILQ_HEAD(mp_arena, mem_arena);
+
+typedef struct mem_block {
```

Comment:
Most likely you don't need to expose it to malloc's user. If so please put it into `memory_allocator.c`.